packages/create-catalyst/src/utils/config.ts

```diff
@@ -0,0 +1,100 @@
+import { parse as parseTOML, stringify as stringifyTOML } from '@iarna/toml';
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'fs';
+import { dirname, join } from 'path';
+
+interface CatalystConfig {
+  auth?: {
+    storeHash?: string;
+    accessToken?: string;
+  };
+}
+
+interface TomlRecord {
+  [key: string]: string | number | boolean | TomlRecord | TomlRecord[];
+}
+
+function isRecord(value: unknown): value is Record<string, unknown> {
+  return typeof value === 'object' && value !== null;
+}
+
+function isCatalystConfig(obj: unknown): obj is CatalystConfig {
+  if (!isRecord(obj)) {
+    return false;
+  }
+
+  if ('auth' in obj) {
+    if (!isRecord(obj.auth)) {
+      return false;
+    }
+
+    const { storeHash, accessToken } = obj.auth;
+
+    return (
+      (storeHash === undefined || typeof storeHash === 'string') &&
+      (accessToken === undefined || typeof accessToken === 'string')
+    );
+  }
+
+  return true;
+}
+
+export class Config {
+  private configPath: string;
+  private config: CatalystConfig;
+
+  constructor(projectDir: string) {
+    this.configPath = join(projectDir, '.catalyst');
+    this.config = this.read();
+  }
+
+  getAuth(): { storeHash?: string; accessToken?: string } {
+    return this.config.auth ?? {};
+  }
+
+  setAuth(storeHash: string, accessToken: string): void {
+    this.config.auth = { storeHash, accessToken };
+    this.save();
+  }
+
+  save(): void {
+    const configObj: TomlRecord = {};
+
+    if (this.config.auth?.storeHash && this.config.auth.accessToken) {
+      configObj.auth = {
+        storeHash: this.config.auth.storeHash,
```

Comment:
Will the same store hash be in the main .env file and this config? If it is, do we need the store hash in both?

Reply:
The utility of the `.catalyst` file is you can regenerate the .env.local, so it makes sense for it to have no dependencies on .env.local